### PR TITLE
Add Arista support

### PR DESCRIPTION
**Pull request by rabin-io**
Add support for Arista EOS devices, since there interfae is very like Cisco, we just duplicated the Cisco template and tweaked few things, and added a new function to lookup AS path for ipv6.

We currently use it on our LG site, but If any one else like to test it and give feedback we will appreciate that, As we like to merge it upstream.

### CODE REVIEW

File: routers/Arista.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
+<?php
```
Comment (respawner) [inline routers/Arista.php]: Filename should be in lowercase.

File: routers/Arista.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    {
+        #$parameter = quote($parameter);
+        $cmd = new CommandBuilder();
+        $cmd->add('show');
+        $cmd->add('ip bgp regex', $parameter);
+
+        return array($cmd);
+    }
+
+    protected function build_aspath_regexp6($parameter): array
```
Comment (respawner) [inline routers/Arista.php]: Consider adding the IPv6 specific AS path regexp command to the existing function. You can take a look at the [`juniper.php`](https://github.com/respawner/looking-glass/blob/master/routers/juniper.php#L44) file as an example.

File: routers/bird.php [hunk 3]
```diff
@@ -69,6 +69,11 @@ protected function build_aspath_regexp($parameter) {
   protected function build_as($parameter) {
     return $this->get_aspath($parameter);
   }
+
```
Comment (respawner) [inline routers/bird.php]: Not required if the IPv6 specific AS path regexp function is removed and its logic included in the existing AS path function.

File: routers/cisco.php [hunk 4]
```diff
@@ -110,6 +110,11 @@ protected function build_traceroute($parameter) {
 
     return array($cmd);
   }
+
```
Comment (respawner) [inline routers/cisco.php]: Not required if the IPv6 specific AS path regexp function is removed and its logic included in the existing AS path function.

File: routers/cisco.php [hunk 5]
```diff
@@ -110,6 +110,11 @@ protected function build_traceroute($parameter) {
...
+    protected function build_aspath_regexp6($parameter)
```
Comment (respawner) [inline routers/cisco.php]: Not required if the IPv6 specific AS path regexp function is removed and its logic included in the existing AS path function.

File: routers/extreme_netiron.php [hunk 6]
```diff
@@ -162,6 +162,11 @@ protected function build_traceroute($parameter) {
 
     return array($cmd);
   }
+
```
Comment (respawner) [inline routers/extreme_netiron.php]: Not required if the IPv6 specific AS path regexp function is removed and its logic included in the existing AS path function.

File: routers/juniper.php [hunk 7]
```diff
@@ -106,6 +106,11 @@ protected function build_traceroute($parameter) {
 
     return array($cmd);
   }
+
```
Comment (respawner) [inline routers/juniper.php]: Not required if the IPv6 specific AS path regexp function is removed and its logic included in the existing AS path function.

File: routers/mikrotik.php [hunk 8]
```diff
@@ -131,6 +131,11 @@ protected function build_traceroute($parameter) {
 
     return array($cmd);
   }
+
```
Comment (respawner) [inline routers/mikrotik.php]: Not required if the IPv6 specific AS path regexp function is removed and its logic included in the existing AS path function.

File: routers/openbgpd.php [hunk 9]
```diff
@@ -74,6 +74,11 @@ protected function build_as($parameter) {
 
     return array($cmd);
   }
+
```
Comment (respawner) [inline routers/openbgpd.php]: Not required if the IPv6 specific AS path regexp function is removed and its logic included in the existing AS path function.

File: routers/quagga.php [hunk 10]
```diff
@@ -62,6 +62,11 @@ protected function build_as($parameter) {
     $parameter = '^'.$parameter.'_';
     return $this->build_aspath_regexp($parameter);
   }
+
```
Comment (respawner) [inline routers/quagga.php]: Not required if the IPv6 specific AS path regexp function is removed and its logic included in the existing AS path function.

File: routers/router.php [hunk 11]
```diff
@@ -21,6 +21,7 @@
 
 require_once('includes/config.defaults.php');
 require_once('config.php');
+require_once 'Arista.php';
```
Comment (respawner) [inline routers/router.php]: Lowercase filename and add parenthesis to match code style.

File: routers/router.php [hunk 12]
```diff
@@ -146,6 +147,8 @@ protected function get_routing_table_name($ip_version = 'ipv6') {
 
   protected abstract function build_aspath_regexp($parameter);
 
+  abstract protected function build_aspath_regexp6($parameter);
```
Comment (respawner) [inline routers/router.php]: Not required if the IPv6 specific AS path regexp function is removed and its logic included in the existing AS path function.

File: routers/router.php [hunk 13]
```diff
@@ -166,6 +169,12 @@ private function build_commands($command, $parameter) {
         }
         return $this->build_aspath_regexp($parameter);
 
+      case 'as-path-regex6':
```
Comment (respawner) [inline routers/router.php]: Not required if the IPv6 specific AS path regexp function is removed and its logic included in the existing AS path function.

File: routers/Arista.php (hunk 1 above)
Comment (rabin-io) [reply routers/Arista.php]: done

File: routers/router.php (hunk 11 above)
Comment (rabin-io) [reply routers/router.php]: done

**Comment by rabin-io** [comment]
Hi,

Just edited my branch based on your comments,
the diff is much smaller, as it just touch the router.php file and add the arista file.

**Pull request merged**